### DISCUSSION TITLE
How to get nested API JSON result from Collection?

### DISCUSSION THREAD

**Discussion by suhaimim** [directus/directus#9967] [Archive]
Hi,

This is the API result for each collections that has a relationship from one to the other (many-to-one) cascaded.
Is there any ways to get the JSON result that already linked to the relations all together? 
Is there any sample if this could be done by custom API creations?

![image](https://user-images.githubusercontent.com/35832684/142788532-29b6d30a-ab69-4fbc-9284-c2d9ddf2f65e.png)

**Comment by azrikahar** [comment]
You can achieve this using the `fields` query parameter, more info in the docs: https://docs.directus.io/reference/query/#fields

**Reply by suhaimim** [reply]
Apologise for raised back to the unaswered mark.
Please refer my following post, I couldnt find the guide on how to use the fields query parameter as per advised above.

**Comment by suhaimim** [comment]
May I know how to apply this into the filtering box?

![image](https://user-images.githubusercontent.com/35832684/142977313-39747e76-59c4-4165-a758-adda19e59962.png)

![image](https://user-images.githubusercontent.com/35832684/142977248-eab0f487-3e2f-406d-a3e4-d02f4e3a1107.png)

![image](https://user-images.githubusercontent.com/35832684/142977356-52e87c82-cfa9-4c3d-a86f-2e12e4d864fa.png)

![image](https://user-images.githubusercontent.com/35832684/142977514-e5813148-cbc9-4f2c-8536-0a7f4b9ff9ab.png)

**Reply by rijkvanzanten** [reply]
I'm not sure what you're asking for exactly.. The limit example in the screenshot is from something totally different than permissions. What are you trying to do?

**Comment by alexvdvalk** [comment]
You can use `?fields=*.*` or more specifically to your example: /items/cities?fields=id,cityName,statesId(statesName,...)

**Reply by mayankgp21** [reply]
Hi Alex, we have the enterprise license for directus and also reached out on support mailer.`?fields=*.*` is not helping us in our use case. We have a nested json schema and are trying to represent that in the form of separate directus collections. it does shows us the id of child objects but not the full object. also the issue is that we don't know the level of nesting will be there beforehand hence we don't know how many times we need to apply `*.*` pattern.
